docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst

```diff
@@ -0,0 +1,326 @@
+- ``ArgSide`` must be one of:
+   - ``KokkosBatched::Side::Left`` to apply the transformation from the left
+   - ``KokkosBatched::Side::Right`` to apply the transformation from the right
+- ``uViewType`` must be a rank-1 view containing the Householder vector
+- ``tauViewType`` must be a scalar or a rank-0 view containing the Householder scalar
+- ``AViewType`` must be a rank-2 view representing the matrix to transform
+- ``wViewType`` must be a rank-1 view with sufficient workspace for the computation
+- All views must be accessible in the execution space
+
+Example
+=======
+
+.. code-block:: cpp
+
+    #include <Kokkos_Core.hpp>
+    #include <KokkosBatched_ApplyHouseholder_Decl.hpp>
+    
+    using execution_space = Kokkos::DefaultExecutionSpace;
+    using memory_space = execution_space::memory_space;
+    
+    // Scalar type to use
+    using scalar_type = double;
+    
+    int main(int argc, char* argv[]) {
+      Kokkos::initialize(argc, argv);
+      {
+        // Matrix dimensions
+        int m = 5;  // Number of rows
+        int n = 3;  // Number of columns
+        
+        // Create matrices and vectors
+        Kokkos::View<scalar_type**, Kokkos::LayoutRight, memory_space> A("A", m, n);
+        Kokkos::View<scalar_type*, memory_space> u("u", m);   // Householder vector
+        Kokkos::View<scalar_type, memory_space> tau("tau");   // Householder scalar
+        Kokkos::View<scalar_type*, memory_space> w("w", n);   // Workspace
+        
+        // Initialize on host
+        auto A_host = Kokkos::create_mirror_view(A);
+        auto u_host = Kokkos::create_mirror_view(u);
+        auto tau_host = Kokkos::create_mirror_view(tau);
+        
+        // Initialize A with recognizable pattern
+        for (int i = 0; i < m; ++i) {
+          for (int j = 0; j < n; ++j) {
+            A_host(i, j) = (i + 1) * 10 + (j + 1);
+          }
+        }
+        
+        // Initialize Householder vector (first element is 1.0, rest are zeros by convention)
+        u_host(0) = 1.0;
+        for (int i = 1; i < m; ++i) {
+          u_host(i) = 0.5 * i;
+        }
+        
+        // Set tau
+        tau_host() = 0.5;
+        
+        // Copy to device
+        Kokkos::deep_copy(A, A_host);
+        Kokkos::deep_copy(u, u_host);
+        Kokkos::deep_copy(tau, tau_host);
+        
+        // Save a copy of the original matrix for verification
+        Kokkos::View<scalar_type**, Kokkos::LayoutRight, memory_space> A_orig("A_orig", m, n);
+        Kokkos::deep_copy(A_orig, A);
+        
+        // Apply Householder transformation from the left
+        Kokkos::parallel_for(1, KOKKOS_LAMBDA(const int i) {
+          KokkosBatched::SerialApplyHouseholder<KokkosBatched::Side::Left>
+            ::invoke(u, tau, A, w);
+        });
+        
+        // Copy results back to host
+        Kokkos::deep_copy(A_host, A);
+        
+        // Verify results: Manually compute H*A and compare
```

Comment:
This is not useful for an example, this is already implemented in our unit-tests with proper checks